reporthandling/results/v1/reportsummary/summarydetails_test.go

```diff
@@ -326,3 +329,205 @@ func TestSummaryDetails_GetControlsSeverityCounters(t *testing.T) {
 		})
 	}
 }
+
+//go:embed testdata/summaryDetails.json
+var summaryDetailsBytes []byte
+
+//go:embed testdata/allResourcesResults.json
+var allResourcesResultsBytes []byte
+
+func setUpSummaryDetails() (*SummaryDetails, error) {
+	summaryDetails := &SummaryDetails{}
+	if err := json.Unmarshal(summaryDetailsBytes, summaryDetails); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal summaryDetailsBytes: %v", err)
+	}
+
+	allResourcesResults := map[string]resourcesresults.Result{}
+	if err := json.Unmarshal(allResourcesResultsBytes, &allResourcesResults); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal allResourcesResults: %v", err)
+	}
+
+	for i := range allResourcesResults {
+		t := allResourcesResults[i]
+		summaryDetails.AppendResourceResult(&t)
+	}
+
+	summaryDetails.InitResourcesSummary(nil)
+
+	return summaryDetails, nil
+}
+func TestSummaryDetails_Counters(t *testing.T) {
+
+	summaryDetails, err := setUpSummaryDetails()
+	if err != nil {
+		t.Fatalf("failed to unmarshal allResourcesResults: %v", err)
+	}
+
+	// testing counters
+	assert.Equal(t, 93, summaryDetails.StatusCounters.All())
+	assert.Equal(t, 4, summaryDetails.StatusCounters.Passed())
+	assert.Equal(t, 9, summaryDetails.StatusCounters.Failed())
+	assert.Equal(t, 80, summaryDetails.StatusCounters.Skipped())
```

Comment:
In general, I agree with you, we have many other tests with local structs and smaller numbers that make more sense.
Here, however, we are testing the E2E counters aggregation process, as you can tell, I embedded the objects because they are not small.
I want to have full coverage since this is what we missed with the latest bug. We made a change that the smaller tests did not catch. This is not good, and I should definitely add more tests and use cases, but for now, with the urgency, I have to deliver this fix I believe this will do.